spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BrokerEventListener.java

```diff
@@ -0,0 +1,174 @@
+
+	private int phase;
+
+	private boolean autoStartup = true;
+
+	private volatile boolean running;
+
+	private volatile boolean bindingsFailed;
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	public BrokerEventListener(ConnectionFactory connectionFactory, String... eventKeys) {
+		this(new DirectMessageListenerContainer(connectionFactory), true, eventKeys);
+	}
+
+	public BrokerEventListener(AbstractMessageListenerContainer container, String... eventKeys) {
+		this(container, false, eventKeys);
+	}
+
+	private BrokerEventListener(AbstractMessageListenerContainer container, boolean ownContainer, String... eventKeys) {
+		Assert.notNull(container, "listener container cannot be null");
+		Assert.isTrue(!ObjectUtils.isEmpty(eventKeys), "At least one event key is required");
+		this.container = container;
+		this.container.setQueues(this.eventQueue);
+		this.container.setMessageListener(this);
+		this.eventKeys = Arrays.copyOf(eventKeys, eventKeys.length);
+		this.container.getConnectionFactory().addConnectionListener(this);
+		this.admin = new RabbitAdmin(this.container.getConnectionFactory());
+		this.ownContainer = ownContainer;
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		this.applicationEventPublisher = applicationEventPublisher;
+	}
+
+	public boolean isBindingsFailed() {
+		return this.bindingsFailed;
+	}
+
+	@Override
+	public synchronized void start() {
+		if (!this.running) {
+			if (this.ownContainer) {
+				this.container.start();
+			}
+			this.running = true;
+		}
+	}
+
+	@Override
+	public synchronized void stop() {
+		if (this.running) {
+			if (this.ownContainer) {
+				this.container.stop();
+			}
+			this.running = false;
+		}
+	}
+
+	@Override
+	public boolean isRunning() {
+		return this.running;
+	}
+
+	@Override
+	public int getPhase() {
+		return this.phase;
+	}
+
+	public void setPhase(int phase) {
+		this.phase = phase;
+	}
+
+	@Override
+	public boolean isAutoStartup() {
+		return this.autoStartup;
+	}
+
+	public void setAutoStartup(boolean autoStartup) {
+		this.autoStartup = autoStartup;
+	}
+
+	@Override
+	public void stop(Runnable callback) {
+		stop();
+		callback.run();
+	}
+
+	@Override
+	public void onMessage(Message message) {
+		if (this.applicationEventPublisher != null) {
+			this.applicationEventPublisher.publishEvent(new BrokerEvent(this, message.getMessageProperties()));
+		}
+	}
+
+	@Override
+	public void onCreate(Connection connection) {
+		TopicExchange exchange = new TopicExchange("amq.rabbitmq.event");
+		try {
+			this.admin.declareQueue(this.eventQueue);
+			Arrays.stream(this.eventKeys).forEach(k -> {
+				Binding binding = BindingBuilder.bind(this.eventQueue).to(exchange).with(k);
+				this.admin.declareBinding(binding);
+			});
+		}
+		catch (Exception e) {
+			logger.error("failed to declare event queue/bindings - is the plugin enabled?");
+			this.bindingsFailed = true;
```

Comment:
Shouldn't we store a failure message as well for case `bindingsFailed` is really `true`.